### ISSUE TITLE
0.45 loses parameter indentation after closure

### ISSUE THREAD

**Issue by jshier** [nicklockwood/SwiftFormat#681]
There seems to be a change in behavior in the indentation of parameters after a closure parameter. This code used the automatic Xcode indentation:

```swift
someController = SomeController(action: { [unowned self] in
             self.value = true; self.actionCalled()
}
                                otherParameter: parameter)
```

0.45 removes the indentation of the `otherParameter`:

```swift
someController = SomeController(action: { [unowned self] in
             self.value = true; self.actionCalled()
},
otherParameter: parameter)
```

This seems like a bug.

Edit: This may be related to Xcode 12, as its indenter seems to prefer the newly formatted version, while 11.6 aligns the parameters. Or it could be coincidence, as 11.6 is my selected version.

**Comment by nicklockwood** [comment]
It's a coincidence, as I've not deliberately updated the `--xcodeindentation` behavior for Xcode 12.

**Comment by nicklockwood** [comment]
@jshier given the changes in Xcode 12, how would you expect this to behave? Should it

1) use Xcode 11-style indenting until Xcode 12 is out of beta and then switch
2) switch to Xcode 12 style now
3) alter the Xcode indentation mode based on the `--swiftversion`
4) add a new `--xcodeversion` option to control the behavior?

**Comment by jshier** [comment]
I'm pretty sure Xcode 12's behavior is a bug and I'm going to make a report to that affect. I'd like to keep the previous behavior (and an update to indent the closure indentation would be great too).

**Comment by nicklockwood** [comment]
@jshier just to be clear, how I intended this to work (i.e. the behavior with `--xcodeindentation false`) should be:

```swift
someController = SomeController(action: { [unowned self] in
                                    self.value = true; self.actionCalled()
                                },
                                otherParameter: parameter)
```

For the `--xcodeindentation true` case, are you saying you would like the behavior to be:

```swift
someController = SomeController(action: { [unowned self] in
    self.value = true; self.actionCalled()
},
                                otherParameter: parameter)
```

or something else?

**Comment by jshier** [comment]
Yes, those cases make sense. The second example matches what I see with Xcode 11.6. I'm not really familiar with the impact of `xcodeindentation`.

**Comment by nicklockwood** [comment]
The idea of the `--xcodeindentation true` option is to make SwiftFormat behave like Xcode (excluding cases where Xcode's behavior is clearly a bug). With the  `--xcodeindentation false`, the formatting is just whatever I think looks good :-)

**Comment by nicklockwood** [comment]
@jshier fixed in 0.45.2.

**Comment by jshier** [comment]
I think this was fixed in 0.45.2, but seems to have regressed in a newer release. Formatting Alamofire on `master` produces several instances of parameter indentation being removed. For example, in [UploadTests](https://github.com/Alamofire/Alamofire/blob/master/Tests/UploadTests.swift#L282) this:

```swift
        AF.upload(multipartFormData: { multipartFormData in
            multipartFormData.append(uploadData, withName: "upload_data")
            formData = multipartFormData
        },
                  to: urlString)
            .response { resp in
                response = resp
                expectation.fulfill()
            }
```

Is reformatted to this:

```swift
        AF.upload(multipartFormData: { multipartFormData in
            multipartFormData.append(uploadData, withName: "upload_data")
            formData = multipartFormData
        },
         to: urlString)
            .response { resp in
                response = resp
                expectation.fulfill()
            }
```

Notice the removal of the indentation for the `to:` parameter.

**Comment by nicklockwood** [comment]
@jshier interestingly, it seems to be a bug related to the `--closingparen same-line` option. Without that, it formats as:

```swift
AF.upload(multipartFormData: { multipartFormData in
                      multipartFormData.append(uploadData, withName: "upload_data")
                      formData = multipartFormData
                  },
                  to: urlString)
            .response { resp in
                response = resp
                expectation.fulfill()
            }
```

Which I would consider correct.

**Comment by nicklockwood** [comment]
@jshier fixed in 0.46.0.
